tools/db_import/config.yml

```diff
@@ -0,0 +1,267 @@
+---
+snippets:
+  extract_model_name: |
+    function(full_model_name) std.split(std.split(full_model_name, 'TF')[0], 'PT')[0]
+  extract_batch_size: |
+    function(full_model_name) local parts = std.split(full_model_name, 'Batch'); if std.length(parts) == 2 then parts[1] else null
+  extract_framework: |
+    function(full_model_name) if 0 < std.length(std.findSubstr('TF', full_model_name)) then 'TF' else if 0 < std.length(std.findSubstr('PT', full_model_name)) then 'PT' else null
+  loadJson: |
+    function(filename) std.parseJson(std.native('readFile')(filename))
+  loadJsonFromFirstFoundFile: |
+    function(filepaths) std.parseJson(std.native('tryReadingFilesUntilSuccess')(filepaths))
+  loadCsv: |
+    function(filename) std.native('parseCsv')(std.native('readFile')(filename))
+  wrapInResultRowForDbImport: |
+    function(rows) [{ results: std.toString(row) } for row in rows]
+  parseNumber: |
+    function(str) std.native('parseNumber')(str)
+  getFilepathCapture: |
+    function(name) std.parseJson(std.extVar('filepath_captures'))[name]
+  getAllFilepathCaptures: |
+    function() std.parseJson(std.extVar('filepath_captures'))
+  timestampToIso8601: |
+    function(timestamp) std.native('timestampToIso8601')(timestamp)
+
+common:
+  common_sql_condition: &common_sql_condition |
+    SELECT 1
+    FROM `{dataset}.{table}`
+    WHERE JSON_VALUE(results.trigger.bucket_name) = @bucket_name
+    AND   JSON_VALUE(results.trigger.config)      = @config
+    AND   JSON_VALUE(results.trigger.prefix)      = @prefix
+    LIMIT 1;
+
+  # We use this query to delete all the data from a given source bucket
+  common_sql_delete: &common_sql_delete |
+    DELETE FROM `{dataset}.{table}`
+    WHERE JSON_VALUE(results.trigger.bucket_name) = @bucket_name
+
+  # We use this query to determine if there is already any data from a given bucket in the DB
+  common_sql_data_present: &common_sql_data_present |
+    SELECT 1
+    FROM `{dataset}.{table}`
+    WHERE JSON_VALUE(results.trigger.bucket_name) = @bucket_name
+    LIMIT 1;
+
+table_schemas:
+  benchmarking_experiments.oobi_v7: results:JSON
+  
+cloud_functions:
+  postsubmit: &postsubmit
+    bucket_name: iree-github-actions-postsubmit-artifacts
+    cloud_function_name: oobi-postsubmit-importer-v7
+    table_name: benchmarking_experiments.oobi_v7
+    sql_delete: *common_sql_delete
+    sql_data_present: *common_sql_data_present
+    service_account: oobi-db-import
+    rules:
+    - filepath_regex: "^(?P<fullpath>(?P<prefix>(?P<github_action_run_id>[0-9]+)/(?P<github_action_run_attempt>[0-9]+))/benchmark-results/benchmark-results-(?P<config>.*)\\.json)$"
+      sql_condition: *common_sql_condition
+      result: |
+        // Import helper functions
```

Comment:
This seems to be a big chunk of code. Is it possible to put them in a file (and we can get proper syntax highlights)

Reply:
Probably. Let me explore this option!